clang/test/CIR/CodeGen/bitfield-ops.c

```diff
@@ -1,29 +1,30 @@
-// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=BEFORE
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o - 2>&1 | FileCheck %s
 
+// CHECK: !ty_22S22 = !cir.struct<struct "S" {!cir.int<u, 32>, !cir.int<u, 32>, !cir.int<u, 16>, !cir.int<u, 32>} #cir.record.decl.ast>
 typedef struct {
   int a : 4;
   int b : 27;
   int c : 17;
   int d : 2;
   int e : 15;
-  unsigned f; 
-} S; 
+  unsigned f;
+} S;
 
-// BEFORE: cir.func {{.*@store_field}}
-// BEFORE:   [[TMP0:%.*]] = cir.alloca !ty_22S22, cir.ptr <!ty_22S22>, ["s"] 
-// BEFORE:   [[TMP1:%.*]] = cir.const(#cir.int<3> : !s32i) : !s32i
-// BEFORE:   [[TMP2:%.*]] = cir.get_member [[TMP0]][2] {name = "e"} : !cir.ptr<!ty_22S22> -> !cir.ptr<!u16i>
-// BEFORE:   [[TMP3:%.*]] = cir.set_bitfield [[TMP1]], [[TMP2]](#e) : (!s32i, !cir.ptr<!u16i>) -> !s32i
+// CHECK: cir.func {{.*@store_field}}
+// CHECK:   [[TMP0:%.*]] = cir.alloca !ty_22S22, cir.ptr <!ty_22S22>, ["s"]
+// CHECK:   [[TMP1:%.*]] = cir.const(#cir.int<3> : !s32i) : !s32i
+// CHECK:   [[TMP2:%.*]] = cir.get_member [[TMP0]][2] {name = "e"} : !cir.ptr<!ty_22S22> -> !cir.ptr<!u16i>
+// CHECK:   [[TMP3:%.*]] = cir.set_bitfield [[TMP1]], [[TMP2]](#bfi_e) : (!s32i, !cir.ptr<!u16i>) -> !s32i
 void store_field() {
   S s;
   s.e = 3;
 }
 
-// BEFORE: cir.func {{.*@load_field}}
-// BEFORE:   [[TMP0:%.*]] = cir.alloca !cir.ptr<!ty_22S22>, cir.ptr <!cir.ptr<!ty_22S22>>, ["s", init] 
-// BEFORE:   [[TMP1:%.*]] = cir.load [[TMP0]] : cir.ptr <!cir.ptr<!ty_22S22>>, !cir.ptr<!ty_22S22>
-// BEFORE:   [[TMP2:%.*]] = cir.get_member [[TMP1]][1] {name = "d"} : !cir.ptr<!ty_22S22> -> !cir.ptr<!u32i>
-// BEFORE:   [[TMP3:%.*]] = cir.get_bitfield [[TMP2]](#d) : !cir.ptr<!u32i> -> !s32i
+// CHECK: cir.func {{.*@load_field}}
+// CHECK:   [[TMP0:%.*]] = cir.alloca !cir.ptr<!ty_22S22>, cir.ptr <!cir.ptr<!ty_22S22>>, ["s", init]
+// CHECK:   [[TMP1:%.*]] = cir.load [[TMP0]] : cir.ptr <!cir.ptr<!ty_22S22>>, !cir.ptr<!ty_22S22>
+// CHECK:   [[TMP2:%.*]] = cir.get_member [[TMP1]][1] {name = "d"} : !cir.ptr<!ty_22S22> -> !cir.ptr<!u32i>
+// CHECK:   [[TMP3:%.*]] = cir.get_bitfield [[TMP2]](#bfi_d) : !cir.ptr<!u32i> -> !s32i
```

Comment:
What I meant in the previous comments: please add a CHECK for the alias definition of `#bfi_d`, both in the tests and in the operation documentation.

Reply:
got it!